### PR TITLE
[SYCLomatic] Implement migration for 21 asm instructions

### PR DESCRIPTION
**Pull request by yihwang-nv**

### CODE REVIEW

**Review by wenhuiNi — APPROVED** [review]

**Comment by yihwang-nv** [comment]
Depends on https://github.com/oneapi-src/SYCLomatic/pull/1090

File: clang/test/dpct/asm_int.cu [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+__global__ void cnot() {
+  uint16_t u16;
+  uint32_t u32;
+  uint64_t u64;
+
+  // CHECK: u16 = 1 == 0;
+  asm("cnot.b16 %0, %1;" : "=r"(u16) : "r"(1));
+
+  // CHECK: u32 = 1 == 0;
+  asm("cnot.b32 %0, %1;" : "=r"(u32) : "r"(1));
```
Comment (zhimingwang36) [inline clang/test/dpct/asm_int.cu]: seem all the lit case use constant value as argument, could u also add variable as argument？

Comment (yihwang-nv) [reply clang/test/dpct/asm_int.cu]: done

File: clang/lib/DPCT/AsmMigration.cpp [hunk 2]
```diff
@@ -794,6 +835,504 @@ class SYCLGen : public SYCLGenBase {
...
+
+  bool handle_add(const InlineAsmInstruction *Inst) override {
+    return HandleAddSub(Inst);
+  }
+
+  bool handle_sub(const InlineAsmInstruction *Inst) override {
+    return HandleAddSub(Inst);
+  }
+
+  bool HandleMulMad(const InlineAsmInstruction *Inst) {
```
Comment (ziranzha) [inline clang/lib/DPCT/AsmMigration.cpp]: Should we split `mul` and `mad`?

Comment (yihwang-nv) [reply clang/lib/DPCT/AsmMigration.cpp]: done

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**